### PR TITLE
MAT-5552: add rate aggregation

### PR DESCRIPTION
**Pull request by chubert-sb**
## MADiE PR

Jira Ticket: [MAT-5552](https://jira.cms.gov/browse/MAT-5552)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [x] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [x] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [x] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.tsx [hunk 1]
```diff
@@ -1,7 +1,175 @@
...
+import React, { useState, useEffect } from "react";
+import { useFormik } from "formik";
+import { Measure } from "@madie/madie-models";
+import {
+  measureStore,
+  checkUserCanEdit,
+  routeHandlerStore,
+} from "@madie/madie-util";
+import MetaDataWrapper from "../../../editMeasure/details/MetaDataWrapper";
+import { QDMMeasureSchemaValidator } from "../../../../validations/QDMMeasureSchemaValidator";
```
Comment (RohitKandimalla) [inline src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.tsx]: Looks like we don't need any validations

**Review by mcmcphillips — DISMISSED** [review]

File: src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.tsx [hunk 2]
```diff
@@ -1,7 +1,175 @@
...
+      isValid={formik.isValid}
+      handleSubmit={formik.handleSubmit}
+      onCancel={onCancel}
+    >
+      <form
+        id="measure-reporting-form"
+        onSubmit={formik.handleSubmit}
+        data-testid="reporting-form"
+      >
+        <div id="reporting" data-testid="reporting">
```
Comment (RohitKandimalla) [inline src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.tsx:120]: Do you think, we could reuse the existing reporting tab from QiCore by making it a separate component? Unless the functionality between these models is different.

File: src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.tsx [hunk 3]
```diff
@@ -1,7 +1,175 @@
...
+          <div className="left">
+            <FieldLabel htmlFor="rate-aggregation" id="rate-aggregation-label">
+              Rate Aggregation
+            </FieldLabel>
+            <FieldSeparator>
+              <FieldInput
+                value={formik.values.rateAggregation}
```
Comment (RohitKandimalla) [inline src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.tsx]: Correct me if I am wrong, but` {...formik.getFieldProps("rateAggregation")}` should already be handling the value.

File: src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+  cqlLibraryName: "TestCqlLibraryName",
+  ecqmTitle: "ecqmTitle",
+  measurementPeriodStart: "01/01/2022",
+  measurementPeriodEnd: "12/02/2022",
+  createdBy: "john doe",
+  measureSetId: "testMeasureId",
+  acls: [{ userId: "othertestuser@example.com", roles: ["SHARED_WITH"] }], //#nosec
+} as unknown as Measure;
+
+jest.mock("@madie/madie-editor", () => ({
```
Comment (RohitKandimalla) [inline src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx]: Do we need to mock Editor?

File: src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    const rateAggregation = getByTestId(
+      "rateAggregationText"
+    ) as HTMLInputElement;
+
+    fireEvent.change(rateAggregation, {
+      target: { value: "Test" },
+    });
+    expect(rateAggregation.value).toBe("Test");
+
+    const cancelButton = getByTestId("cancel-button");
```
Comment (RohitKandimalla) [inline src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx]: RTL specifies a priority and it is advised to use getByRole as a top priority, I think we can use that in here 
https://testing-library.com/docs/queries/about#priority

    const cancelButton = screen.getByRole("button", {
      name: "Discard Changes",
    });

File: src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+    expect(cancelButton).toBeInTheDocument();
+    await waitFor(() => expect(cancelButton).toBeEnabled());
+    act(() => {
+      fireEvent.click(cancelButton);
```
Comment (RohitKandimalla) [inline src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx]: FireEvent is already wrapped in act, So we don;t have to explicitly wrap it in act

**Review by RohitKandimalla — CHANGES_REQUESTED** [review]

File: src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+import { act } from "react-dom/test-utils";
+import useMeasureServiceApi, {
+  MeasureServiceApi,
+} from "../../../../api/useMeasureServiceApi";
+import { Measure } from "@madie/madie-models";
+import QDMReporting from "./QDMReporting";
+
+const mockHistoryPush = jest.fn();
+
+jest.mock("react-router-dom", () => ({
```
Comment (RohitKandimalla) [inline src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx]: not sure if routeHandlerStore requires this to be mocked, but I don't see a direct usage of react router

**Review by RohitKandimalla — CHANGES_REQUESTED** [review]

File: src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx (hunk 4 above)
Comment (mcmcphillips) [reply src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx]: Jest does not understand anything that we're getting across the browser.

single spa is connected at the browser, so everything outside of your src or node_modules needs to be fully mocked like this.

Comment (RohitKandimalla) [reply src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx]: but we are only rendering the `<QDMReporting />` component, and the editor will not even be a part of the rendered HTML in Jest.

File: src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.tsx (hunk 1 above)
Comment (chubert-sb) [reply src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.tsx]: Oops forgot to remove this. Done.

File: src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.tsx (hunk 2 above)
Comment (chubert-sb) [reply src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.tsx:120]: That's a good point. I didn't really think about it at the time. Maybe? I'm not sure if functionality is planned to be the same or not

File: src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.tsx (hunk 3 above)
Comment (chubert-sb) [reply src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.tsx]: good to know. done

File: src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx (hunk 5 above)
Comment (chubert-sb) [reply src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx]: done

File: src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx (hunk 6 above)
Comment (chubert-sb) [reply src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx]: done

File: src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx (hunk 7 above)
Comment (chubert-sb) [reply src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx]: removed

File: src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.tsx (hunk 2 above)
Comment (RohitKandimalla) [reply src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.tsx:120]: As far as I know, it should function the same way, but we can confirm it from POs

Comment (chubert-sb) [reply src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.tsx:120]: There's MAT-5553 to do Improvement Notation too.

**Review by RohitKandimalla — APPROVED** [review]

**Pull request merged**